### PR TITLE
feat: define versioned application CV artifact schema

### LINKED ISSUE #1779: Group reusable CV and JD artifacts by application
## Problem

The PDF workflow currently stores generated HTML and PDF files directly under `output/`, while JD snapshots live under `jds/` and reports live under `reports/`. When a CV is reused or lightly tailored for a similar JD, the source CV, comparison JD, current JD, generated HTML/PDF, and reuse decision are not grouped together.

This makes it harder to audit which CV version was used for an application and why reuse was chosen.

## Proposal

Use an application-scoped output directory, keyed by the report number and company/role slug, for example:

```text
output/1752-acme-ai-engineer/
  jd/current.md
  jd/previous.md
  cv/source.html
  cv/tailored.html
  cv/tailored.pdf
  reuse-decision.json
```

The implementation should preserve existing PDF/report manifest links and remain backward-compatible with one-off output paths.

## Acceptance criteria

- Reused or lightly tailored CV artifacts are stored together with their comparison inputs.
- The reuse decision records the decision, score, source paths, and changed sections.
- `data/pdf-index.tsv` continues to link the generated PDF and source HTML to the report.
- Existing flat `output/` paths remain supported for backward compatibility.
- Tests cover application-scoped paths and manifest behavior.

### PR DESCRIPTION
**Pull request by luochen211**
## Summary

- Define a versioned, application-scoped artifact layout for reused and tailored CVs.
- Separate source CVs from tailored CV versions.
- Keep current/previous JD snapshots and reuse decisions in the same application bundle.
- Add `npm run application:init` and a reusable path/decision writer.
- Register the system-layer helper and add auto-discovered tests.
- Keep flat `output/` paths supported for one-off PDFs; PDF generation integration is intentionally left for a follow-up PR.

Closes #1779

## Layout

```text
output/{report}-{company}-{role}/
  jd/current.md
  jd/previous.md
  cv/source/original.html
  cv/source/original.pdf
  cv/tailored/v001/cv.html
  cv/tailored/v001/cv.pdf
  cv/tailored/v001/changes.md
  decision/reuse.json
```

## Validation

- `node tests/application-artifacts.test.mjs`
- `node application-artifacts.mjs --report 7 --company "Acme AI" --role "Senior AI Engineer" --version 2 --root /tmp/career-ops-artifacts-test --init`
- `node updater-migration-tests.mjs` — 318 passed, 0 failedn- `node validate-system-paths-coverage.mjs` — 897 tracked system files coveredn- `node test-all.mjs --quick` — 2522 passed, 0 failed, 2 expected warnings

## Dependencies

- Satisfied: #1752 merged and the documented `npm run jd:similarity` command is now present on `main`.

## Summary by CodeRabbit

- **New Features**
  - Added deterministic, application-scoped artifact bundles for job descriptions and CVs (source/tailored), including versioned outputs.
  - Introduced `application:init` to initialize the directory structure and write auditable reuse decision records.
  - Updated the application-scoped PDF workflow to support bundle-relative HTML/PDF paths and stricter reuse behavior when required inputs are missing.
- **Documentation**
  - Updated quick reference and PDF guidance for the new initialization workflow and path abstractions.
- **Tests**
  - Added end-to-end coverage for stable artifact paths, directory creation, reuse decision persistence, and CLI argument validation.
- **Chores**
  - Updated the system updater manifest so the initializer is applied/rolled back with other system components.

### CODE REVIEW

File: application-artifacts.mjs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    console.error(usage());
+    process.exitCode = 1;
+    return;
+  }
+  const paths = applicationArtifactPaths({ reportNum: values.report, company: values.company, role: values.role, version: values.version, root: values.root });
+  if (values.init) ensureApplicationArtifactDirs(paths);
+  console.log(JSON.stringify(paths, null, 2));
+}
+
+if (process.argv[1] && fileURLToPath(import.meta.url) === resolve(process.argv[1])) main();
```
[the remark this answers is not in the document]
Comment (luochen211) [reply application-artifacts.mjs]: Fixed in 88947a5. The CLI entry point now catches async failures, prints a prefixed one-line stderr message, and sets exit code 1. A subprocess regression test verifies there is no stack trace.

File: tests/application-artifacts.test.mjs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  });
+  const decision = JSON.parse(readFileSync(paths.decision.reuse, 'utf8'));
+  if (decision.decision === 'reuse-with-edits' && decision.changed_sections.length === 2) {
+    console.log('  ✅ reuse decisions are recorded beside the artifact bundle');
+  } else {
+    throw new Error(`unexpected reuse decision: ${JSON.stringify(decision)}`);
+  }
+} finally {
+  rmSync(root, { recursive: true, force: true });
+}
```
[the remark this answers is not in the document]
Comment (luochen211) [reply tests/application-artifacts.test.mjs:86]: Covered in 88947a5: invalid report number, invalid version, unsupported decision, non-array changedSections, and punctuation-only slug fallback now have explicit assertions.

**Comment by luochen211** [comment]
Synced with current `main` and resolved the PDF workflow/docs conflicts while preserving main’s HTML builder and fact gate. Removed the unrelated scan-runs diff, added clean CLI failure handling plus all requested validation edge cases, and added explicit coverage that nested application HTML/PDF paths remain repo-relative manifest entries for `data/pdf-index.tsv`. Fresh verification: `node tests/application-artifacts.test.mjs` passes and `node test-all.mjs` reports 1691 passed, 0 failed, 2 expected environment warnings.

**Comment by luochen211** [comment]
Synced this branch with current `main`; no unresolved review threads remain. Fresh verification: `node tests/application-artifacts.test.mjs` passes all 10 checks and `node test-all.mjs` passes 1691 tests with 0 failures (2 expected environment warnings). Ready for review.

**Comment by santifer** [comment]
Seen, @luochen211: a versioned CV artifact schema is contract-defining work, so it gets the architectural review track (labeled). It'll be reviewed after the tracker/data-integrity batch settles, since those decisions constrain this one.

**Comment by luochen211** [comment]
@santifer Synced this architectural-track PR with current `main` and addressed all outstanding review feedback:
- added the below-4.0/5 explicit-override gate
- made `language.output` authoritative for CV output language
- shell-quoted updater pathspecs safely
- preserved current main’s page-budget/fact-gate behavior
- documented the dependency on #1752 for `jd:similarity`

The first CI run exposed a Windows-only path-separator assertion; I replaced it with platform-native `path.join` comparisons and reran everything. Current GitHub checks are all green on Ubuntu, macOS, and Windows, including CodeQL, dependency review, CodeRabbit, and security checks. Local validation: application artifact suite 10/10, updater migration 318/0, path coverage 897 files, and quick suite 2522/0 with 2 expected warnings. No review threads remain unresolved. Ready for architectural review after #1752 lands.

**Comment by luochen211** [comment]
Follow-up: #1752 is now merged, so the `jd:similarity` dependency is satisfied; the PR description has been updated accordingly. After the maintainer sync commit 8407e53 brought the latest `main` into this branch, the new CI run completed fully green on Ubuntu, macOS, and Windows, with CodeQL, CodeRabbit, dependency review, and security checks also passing. The PR remains ready for its architectural review.

**Comment by santifer** [comment]
The architectural one lands, @luochen211: versioned bundles for the application's whole artifact story, with flat output/ paths staying first-class so nothing existing breaks — exactly the coexistence the #1779 green-light asked for. Your bundle-aware jd:similarity integration meshing with the standalone one that merged an hour ago is the kind of collision that resolves into something better than either. Merged. 🚀
